embark-consult.el

```diff
@@ -103,44 +103,84 @@
 (setf (alist-get 'consult-location embark-default-action-overrides)
       #'embark-consult-goto-location)
 
-(defun embark-consult-export-occur (lines)
-  "Create an occur mode buffer listing LINES.
-The elements of LINES are assumed to be values of category `consult-line'."
-  (let ((buf (generate-new-buffer "*Embark Export Occur*"))
-        (mouse-msg "mouse-2: go to this occurrence")
-        last-buf)
+(defun embark-consult--setup-occur-mode (results-writer)
+  "Create a buffer and configure `occur-mode' and call occur results writer"
+  (let ((buf (generate-new-buffer "*Embark Export Occur*")))
     (with-current-buffer buf
-      (dolist (line lines)
-        (pcase-let*
-            ((`(,loc . ,num) (consult--get-location line))
-             ;; the text properties added to the following strings are
-             ;; taken from occur-engine
-             (lineno (propertize (format "%7d:" num)
-                                 'occur-prefix t
-                                 ;; Allow insertion of text at the end
-                                 ;; of the prefix (for Occur Edit mode).
-                                 'front-sticky t
-                                 'rear-nonsticky t
-                                 'occur-target loc
-                                 'follow-link t
-                                 'help-echo mouse-msg))
-             (contents (propertize (embark-consult--strip line)
-                                   'occur-target loc
-                                   'occur-match t
-                                   'follow-link t
-                                   'help-echo mouse-msg))
-             (nl (propertize "\n" 'occur-target loc))
-             (this-buf (marker-buffer loc)))
-          (unless (eq this-buf last-buf)
-            (insert (propertize
-                     (format "lines from buffer: %s\n" this-buf)
-                     'face list-matching-lines-buffer-name-face))
-            (setq last-buf this-buf))
-          (insert (concat lineno contents nl))))
+      (funcall results-writer)
       (goto-char (point-min))
       (occur-mode))
     (pop-to-buffer buf)))
 
+(defun embark-consult-export-location-occur (lines)
+  "Create an occur or grep mode buffer listing LINES.
+The elements of LINES are assumed to be values of category `consult-location'."
+  (funcall
+   #'embark-consult--setup-occur-mode
```

Comment:
Unnecessary funcall here.

Reply:
You're right. I remember you suggested closures and I overlooked that there's an even simpler solution. The fix is on its way.